scipy/integrate/tests/test_quadrature.py

```diff
@@ -397,3 +397,222 @@ def func(x):
+            (
+                # Multi dimensional input for initial
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                None,
+                [[1 / 3], [2 / 3], [1.0]],
+                -1,
+                np.array(
+                    [
+                        [1 / 3, 8 / 3, 9.0, 64 / 3],
+                        [2 / 3, 3.0, 28 / 3, 65 / 3],
+                        [1.0, 10 / 3, 29 / 3, 22.0],
+                    ]
+                ),
+            ),
+            (
+                # Multi dimensional input for dx
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                [[1.0], [2.0], [1.0]],
+                None,
+                None,
+                np.array(
+                    [
+                        [7 / 3, 26 / 3, 21.0],
+                        [14 / 3, 52 / 3, 42.0],
+                        [7 / 3, 26 / 3, 21.0],
+                    ]
+                ),
+            ),
+            (
+                # Multi dimensional values for initial and dx with non-default axis
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                [[1.0, 2.0, 1.0, 5.0]],
+                [[0, 0, 0, 1]],
+                0,
+                np.array(
+                    [
+                        [0, 0, 0, 1],
+                        [1, 8, 9, 81],
+                        [2, 16, 18, 161],
+                    ]
+                ),
+            ),
+        ],
+    )
+    def test_simpson_multi_dx(self, y, dx, initial, axis, int_y):
+        kwargs = {}
+        if dx is not None:
+            kwargs["dx"] = dx
+        if initial is not None:
+            kwargs["initial"] = initial
+        if axis is not None:
+            kwargs["axis"] = axis
+        assert_almost_equal(
+            cumulative_simpson(y, **kwargs),
+            int_y,
+        )
+
+    @pytest.mark.parametrize(
+        "y_func,x,dx,initial,axis,int_y_func",
+        [
+            (
+                # Non-unique values in x
+                lambda y: y**2,
+                np.array([2, 2, 4, 8]),
+                None,
+                8 / 3,
+                -1,
+                lambda y: y**3 / 3,
+            ),
+            (
+                # Multi dimensional, non-unique values in x
+                lambda y: y**2,
+                np.array([[1, 2, 3, 4], [0, 2, 4, 8], [2, 2, 4, 8]]),
+                None,
+                np.array([[1 / 3], [0], [8 / 3]]),
+                -1,
+                lambda y: y**3 / 3,
+            ),
+            (
+                # Multi dimensional, non-unique values in x in non-default axis
+                lambda y: y**2,
+                np.array([[1, 2, 3, 4], [1, 2, 4, 8], [1, 2, 4, 8]]),
+                None,
+                None,
+                0,
+                lambda y: y**3 / 3,
+            ),
+        ],
+    )
+    def test_simpson_exceptions(self, y_func, x, dx, initial, axis, int_y_func):
```

Comment:
How about adding a test for `y.shape[0] < 3`?

Reply:
Agreed, added this test case